js/autocomplete.js

```diff
@@ -5,83 +5,102 @@
+        function submitRelatedResourceForms() {
+          if (0 == newRelatedResourceForms.length) {
+            // Return true to continue with form submit
+            return true;
+          }
 
-          // If no preSubmits fired, then continue with form submit as usual
-          if (count == 0) {
-            return true;  // Go ahead and submit form
+          for (var i = 0; i < newRelatedResourceForms.length; i++) {
+            newRelatedResourceForms[i].submit();
           }
 
-          // Prevent form submit from continuing - final submit will be triggered
-          // by done() function above
+          // Prevent $form.submit() from continuing - final submit will be
+          // triggered by done() function above
           return false;
+        }
+
+        // Submit any related resource forms and wait for response before
+        // submitting the main $form
+        $form.submit(function (event) {
+          return submitRelatedResourceForms();
         });
 
         $('select.form-autocomplete', this).each(createYuiDiv);
 
         // Use $(document).on('loadFunctions') to add function in new rows created with multiRow.js
         $(document).on('loadFunctions', 'select.form-autocomplete', createYuiDiv);
 
-        function createIFrame($src, $dest, uri, targetInput) {
-          // Add hidden <iframe/> (This is the usual behaviour)
-          var $iframe = $('<iframe src="' + uri + '"/>')
+        function createIFrame($src, $dest, uri, targetInputId) {
+          // Add hidden <iframe/>
+          var $iframe = $('<iframe/>')
             .width(0)
             .height(0)
             .css('border', 0);
 
-          // One submit handler for each new choice, use
-          // named function so it can be unbound if choice is
-          // removed
-          var submit = function () {
-            // Increment counter
-            count++;
-
-            // After iframe is submitted, wait for "load" event
-            $iframe.one('load', function () {
-              // Add the resulting page URI (e.g. "/newSubjectName") of the
-              // response to the $bindTo element
-              $dest.val(this.contentWindow.document.location).trigger('change');
-
-              // Submit $form when all preSubmit iframes have submitted, and
-              // reloaded
-              done();
-            });
+          // Wait until iframe contents have loaded so $input and $iframeForm
+          // selectors work
+          $iframe.one('load', function () {
+            // Get iframe > form > input using targetInputId, and parent form
+            var $input = $iframe.contents().find(targetInputId);
+
+            // $($input[0].form) should be faster than $input.closest("form")
+            var $iframeForm = $($input[0].form);
+
+            // Add one submit handler for each new choice. Use a named function
+            // so it can be unbound if the choice is removed
+            var submit = function () {
```

Comment:
It is a bit confusing for me to call the function `submit`, but it's just cosmetic and it may be only me :P

Reply:
Well, it is a function that's bound to the form submit event... 😆 

I'm open to changing the name. Do you find it confusing because it the same handle as the "submit" event?  Would "submitCallback" be less confusing?

Reply:
Heheh, yeah, I think submitCallback is better ;)